src/widget.ts

```diff
@@ -60,26 +122,44 @@ class CodeMirrorWidget extends Widget {
   /**
    * Construct a CodeMirror widget.
    */
-  constructor(model: IEditorViewModel) {
+  constructor(model: IEditorModel) {
     super();
     this.addClass(FILE_BROWSER_CLASS);
-    this._model = model;
     this._editor = CodeMirror(this.node);
-    this.updateMimetype(model.mimetype);
-    this.updateFilename(model.filename);
-    this.updateReadOnly(model.readOnly);
-    this.updateTabSize(model.tabSize);
-    this.updateLineNumbers(model.lineNumbers);
-    this.updateFixedHeight(model.fixedHeight);
-    this.updateText(model.text);
+    this.model = model;
+  }
+
+  /**
+   * Get the editor model.
+   */
+  get model(): IEditorModel {
+    return this._model;
+  }
 
+  /**
+   * Set the editor model, can not be `null`.
+   */
+  set model(value: IEditorModel) {
+    if (value === null) {
```

Comment:
You probably want to support a `null` model, or otherwise use a default-constructed model. If `null` is passed to the constructor, you'll end up with a `null` model.